satellite/marketing/marketingweb/server.go

```diff
@@ -128,20 +170,73 @@ func (s *Server) parseTemplates() (err error) {
 	return nil
 }
 
+// timeConverter formats form time input as time.Time.
+func timeConverter(value string) reflect.Value {
+	v, err := time.Parse("2006-01-02", value)
+	if err != nil {
+		return reflect.Value{}
+	}
+	return reflect.ValueOf(v)
+}
+
+// formToStruct decodes POST form data into a new offer.
+func formToStruct(w http.ResponseWriter, req *http.Request) (o marketing.NewOffer, e error) {
+	err := req.ParseForm()
+	if err != nil {
+		return o, err
+	}
+
+	if err := decoder.Decode(&o, req.PostForm); err != nil {
+		return o, err
+	}
+	return o, nil
+}
+
+// CreateOffer handles requests to create new offers.
+func (s *Server) CreateOffer(w http.ResponseWriter, req *http.Request) {
+	o, err := formToStruct(w, req)
+	if err != nil {
+		s.log.Error("failed to convert form to struct", zap.Error(err))
+		s.serveInternalError(w, req, err)
```

Comment:
Use the correct http error code. This is not a internal server error.

Reply:
If the server fails to convert the form to a struct because a referral admin input invalid form values I wan't that to be displayed on the page `serveInternalError` serves so when they file a ticket and attach that message I know to tell them that they are not inputting correct form values. If they get the 404 page instead that doesn't help me help them figure out what went wrong.

Reply:
The correct error code here would be `400` and you can still show what wasn't parsed correctly.